tools/webdriver/webdriver/bidi/client.py

```diff
@@ -0,0 +1,264 @@
+import asyncio
+import functools
+import json
+
+from urllib.parse import urljoin, urlparse
+from collections import defaultdict
+
+import websockets
+
+
+class BidiException(Exception):
+    def __init__(self, err, msg, stack=None):
+        self.err = err
+        self.msg = msg
+        self.stack = stack
+
+
+class BidiSession:
```

Comment:
Some documentation explaining the lifetime and responsibilities of this class would be great.

And if we can get https://github.com/web-platform-tests/wpt/pull/28900 landed and tools/webdriver under mypy (I volunteer) it would be great to type annotate this from the start.

Reply:
That's now up for review at https://github.com/web-platform-tests/wpt/pull/28913.